var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -91,16 +100,61 @@ class Gcc(AutotoolsPackage):
     patch('piclibs.patch', when='+piclibs')
     patch('gcc-backport.patch', when='@4.7:4.9.2,5:5.3')
 
+    @run_before('autoreconf')
+    def check_languages(self):
+        """Makes sure all requested languages are valid for a
+        specific version of GCC.
+
+        For a list of valid languages for a specific release,
+        run the following command in the GCC source directory:
+
+        .. code-block:: console
+
+           $ grep ^language= gcc/*/config-lang.in
+
+        See https://gcc.gnu.org/install/configure.html
+        """
+        spec = self.spec
+        version = self.version
+
+        # Ada is not currently supported as it requires
+        # an existing Ada compiler to build.
+        if 'languages=ada' in spec:
+            raise InstallError('Ada requires GNAT to install')
+
+        # Support for processing BRIG 1.0 files was added in GCC 7
+        # BRIG is a binary format for HSAIL:
+        # (Heterogeneous System Architecture Intermediate Language).
+        # See https://gcc.gnu.org/gcc-7/changes.html
+        if version < Version('7') and 'languages=brig' in spec:
+            raise InstallError('BRIG is not available before GCC 7')
+
+        # GCC 4.8 added a 'c' language. I'm sure C was always built,
+        # but this is the first version that accepts 'c' as a valid language.
+        if version < Version('4.8') and 'languages=c' in spec:
+            raise InstallError('C is not a valid language before GCC 4.8')
+
+        # GCC 4.6 added support for the Go programming language.
+        # See https://gcc.gnu.org/gcc-4.6/changes.html
+        # See https://golang.org/doc/install/gccgo#Releases
+        if version < Version('4.6') and 'languages=go' in spec:
+            raise InstallError('Go is not available before GCC 4.6')
```

Comment:
I'm not sure why the macos check was there, the mismatch on version number was that 4.7.1 seemed to be the first version capable of bootstrapping the self-hosted go compiler IIRC.